include/psp2/net/adhoc_matching.h

```diff
@@ -0,0 +1,101 @@
+#define _PSP2_NET_ADHOC_MATCHING_H_
+
+#include <vitasdk/build_utils.h>
+#include <psp2/types.h>
+#include <psp2/net/net.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum SceNetAdhocMatchingErrorCode {
+    SCE_NET_ADHOC_MATCHING_ERROR_INVALID_MODE        = 0x80413101,
+    SCE_NET_ADHOC_MATCHING_ERROR_INVALID_PORT        = 0x80413102,
+    SCE_NET_ADHOC_MATCHING_ERROR_INVALID_MAXNUM      = 0x80413103,
+    SCE_NET_ADHOC_MATCHING_ERROR_RXBUF_TOO_SHORT     = 0x80413104,
+    SCE_NET_ADHOC_MATCHING_ERROR_INVALID_OPTLEN      = 0x80413105,
+    SCE_NET_ADHOC_MATCHING_ERROR_INVALID_ARG         = 0x80413106,
+    SCE_NET_ADHOC_MATCHING_ERROR_INVALID_ID          = 0x80413107,
+    SCE_NET_ADHOC_MATCHING_ERROR_ID_NOT_AVAIL        = 0x80413108,
+    SCE_NET_ADHOC_MATCHING_ERROR_NO_SPACE            = 0x80413109,
+    SCE_NET_ADHOC_MATCHING_ERROR_IS_RUNNING          = 0x8041310a,
+    SCE_NET_ADHOC_MATCHING_ERROR_NOT_RUNNING         = 0x8041310b,
+    SCE_NET_ADHOC_MATCHING_ERROR_UNKNOWN_TARGET      = 0x8041310c,
+    SCE_NET_ADHOC_MATCHING_ERROR_TARGET_NOT_READY    = 0x8041310d,
+    SCE_NET_ADHOC_MATCHING_ERROR_EXCEED_MAXNUM       = 0x8041310e,
+    SCE_NET_ADHOC_MATCHING_ERROR_REQUEST_IN_PROGRESS = 0x8041310f,
+    SCE_NET_ADHOC_MATCHING_ERROR_ALREADY_ESTABLISHED = 0x80413110,
+    SCE_NET_ADHOC_MATCHING_ERROR_BUSY                = 0x80413111,
+    SCE_NET_ADHOC_MATCHING_ERROR_ALREADY_INITIALIZED = 0x80413112,
+    SCE_NET_ADHOC_MATCHING_ERROR_NOT_INITIALIZED     = 0x80413113,
+    SCE_NET_ADHOC_MATCHING_ERROR_PORT_IN_USE         = 0x80413114,
+    SCE_NET_ADHOC_MATCHING_ERROR_STACKSIZE_TOO_SHORT = 0x80413115,
+    SCE_NET_ADHOC_MATCHING_ERROR_INVALID_DATALEN     = 0x80413116,
+    SCE_NET_ADHOC_MATCHING_ERROR_NOT_ESTABLISHED     = 0x80413117,
+    SCE_NET_ADHOC_MATCHING_ERROR_DATA_BUSY           = 0x80413118,
+    SCE_NET_ADHOC_MATCHING_ERROR_INVALID_ALIGNMENT   = 0x80413119
+} SceNetAdhocMatchingErrorCode;
+
+typedef enum SceNetAdhocMatchingMode{
+    SCE_NET_ADHOC_MATCHING_MODE_PARENT = 1,
+    SCE_NET_ADHOC_MATCHING_MODE_CHILD,
+    SCE_NET_ADHOC_MATCHING_MODE_P2P,
+} SceNetAdhocMatchingMode;
+
+typedef enum SceNetAdhocMatchingHandlerEventType {
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_HELLO        = 1,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_REQUEST      = 2,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_LEAVE        = 3,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_DENY         = 4,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_CANCEL       = 5,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_ACCEPT       = 6,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_ESTABLISHED  = 7,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_TIMEOUT      = 8,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_ERROR        = 9,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_BYE          = 10,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_DATA         = 11,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_DATA_ACK     = 12,
+    SCE_NET_ADHOC_MATCHING_HANDLER_EVENT_DATA_TIMEOUT = 13
```

Comment:
- have you get these enums via reversing module by yourself?
- doesn't referenced leaked/official sdk?

Reply:
I reversed and re-implemented the full module in https://github.com/Vita3K/Vita3K/pull/3470 although my work is a continuation of another WIP from another user that happens to have this exact enum already. The module behavior does seem to match with the enum names.

Since I can't confirm the source of this enum I can remove it as it's not essential for this header to work properly.

Reply:
yep. please remove them and keep only your own reversing part. we cannot allow official sdk part. unknown source has same risk.

Reply:
done

Reply:
@german77 sorry. something mores.

1. are remain enums safe?
2. please match code convention
    - use hardtab for indentation
    - if you doesn't have special reason, please use `snake_case` for structure field & function arguments (code coventions)
        - for example; `optLen` to be `opt_len`

@Princess-of-Sleeping can you support this review if I late :'(

Reply:
Remaining enums are safe. I did the rest of this header by myself and I also have an usage example that I will upstream once this is merged. I will address the formatting issues in the following minutes.

Reply:
Just as a side note the files in the same folder as this one don't match the convention.